src/overlay/Peer.cpp

```diff
@@ -464,8 +494,39 @@ Peer::sendMessage(StellarMessage const& msg, bool log)
     case SURVEY_RESPONSE:
         getOverlayMetrics().mSendSurveyResponseMeter.Mark();
         break;
+    case SEND_MORE:
+        getOverlayMetrics().mSendSendMoreMeter.Mark();
+        break;
     };
 
+    // Determine if flood messages need to be flow-controlled
+    // Also: place ping responses behind SCP messages to ensure we get accurate
+    // latency measurements
+    bool isFlood = mApp.getOverlayManager().isFloodMessage(msg);
+    if ((isFlood || maybePingResponse) &&
+        mRemoteOverlayVersion >= Peer::FIRST_VERSION_SUPPORTING_FLOW_CONTROL)
+    {
+        // Flow control is not yet set, so don't send anything to the peer,
+        // until we receive the SEND_MORE message
+        if (!mRemotePeerWantsFlowControl)
+        {
+            return;
+        }
+
+        if (*mRemotePeerWantsFlowControl)
+        {
+            mFlowControlledQueue.insert(
```

Comment:
* as discussed, Compare is correct here since it meets the strict weak ordering requirements
* yeah, i just wanted to replace my original two vectors with two deques, but ended up with a sorted data structure as per feedback elsewhere. I think it's actually causing some overhead, so I'll update it to `std::array<std::deque<>>`